### ISSUE TITLE
Repeated `play` execution on 3.9.0

### ISSUE THREAD

**Issue by viralpraxis** [sunny/actor#155]
After upgrading to 3.9.0, I noticed something odd is happening. I managed to simplify it to the following MRE:

```ruby
# frozen_string_literal: true

# some other actor defined somewhere
Another = Class.new(ApplicationActor) do
  output :outcome

  def call
    p "ENTER"

    self.outcome = nil
  end
end

class TestActor < ApplicationActor
  play Another
end
```

Invoking it shows `Another#call` was executed twice:

```
[3] pry(main)> TestActor.result;
"ENTER"
"ENTER"
```

And on 3.8.1:

```
[2] pry(main)> TestActor.result;
"ENTER"
```

I think this is a critical bug -- I was extremely lucky to notice it before deploying to production. In my case it was doubled external API calls..

**Comment by sunny** [comment]
Thank you for bringing this up! Indeed it’s when the last evaluation is `nil` inside `#call`, due to this https://github.com/sunny/actor/pull/152/files#diff-d73c3e509e4af8f87482c4b7d10138db06ae8321256139b8d83e51665c6ac9c9R67 change.

I’m preparing a fix ASAP.

**Comment by sunny** [comment]
Fixed in v3.9.1 🚀 

Thank you for the quick message and investigation 🙏🏻
